Build linux/arm64 images

motoki317 opened the pull request:
Hello,

I noticed that there is no multiplatform support of the driver image, and I thought it'd be nice to have the image built for both linux/amd64 and linux/arm64 platforms.
This change utilizes docker actions to build multiplatform images, which replaces old script in Makefile.

I tested it in my own arm64 environment on OCI A1 instance, it seems to be working.
ref: [package](https://github.com/motoki317/k8s-csi-s3/pkgs/container/k8s-csi-s3), [manifest (values.yaml)](https://github.com/motoki317/manifest/blob/7a35d90de4229554ba01ad83b2d82e3024b633fe/csi-s3/values.yaml)

Thank you in advance.

motoki317 commented:
@vitalif Hi, it would be great if you could take a look at this patch and possibly merge it. Thanks!

vitalif commented:
Hi, why did you remove local makefile? I want to be able to build it locally without github ci.

vitalif commented:
I also think that arm64 csi-s3 can be built without qemu at all because Go allows for very easy cross compilation

motoki317 commented:
> Hi, why did you remove local makefile? I want to be able to build it locally without github ci.

I thought it would be verbose because everything can be built on GitHub Actions, but I guess it's helpful to keep it so you can test it (and publish it) locally.
I dropped the commit in which I removed the Makefile script.

> I also think that arm64 csi-s3 can be built without qemu at all because Go allows for very easy cross compilation

You're right! This also drastically reduces build time, thanks for the suggestion.
I also removed QEMU setup which was not exactly needed in this case.

motoki317 commented:
@vitalif Hi, I fixed the patch following your comments. Would you take a look at this again?

vitalif commented:
Sorry that you have to wait, I think your PR is fine, but I wasn't sure about the multi-architecture support in Yandex container registry. It seems it's not supported yet. So for now it won't work :) Container registry team says they are already working on a fix, so I'll merge it when it will be ready.

motoki317 commented:
Thanks for the update! I'm glad to hear that. I'm in no rush, looking forward to it.

NikoGrano commented:
@vitalif Sorry for pushing, however, do we have an update on this yet? Is there anything we could assist with?

The lack of arm64 is currently blocking me to continue my research 😞.

vitalif commented:
Hi. No, container registry support for "fat manifests" is not rolled out yet. I suppose you can use your own builds by now?)

NikoGrano commented:
@vitalif Yes, I was able to do so. However following images need to be also upgraded (no arm support):

https://github.com/yandex-cloud/k8s-csi-s3/blob/4e410df6e190775355e6afe2fdb5a56b1db3651a/deploy/kubernetes/csi-s3.yaml#L47-L51

Here we need to use image:
```
registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.8.0
```

---

https://github.com/yandex-cloud/k8s-csi-s3/blob/4e410df6e190775355e6afe2fdb5a56b1db3651a/deploy/kubernetes/provisioner.yaml#L77-L80

And here also:

```
registry.k8s.io/sig-storage/csi-provisioner:v3.3.0
```


Anyways `quay.io` images are **deprecated** and we should be using `registry.k8s.io` instead.

space192 commented:
any update on this ?

nidr0x commented:
Any updates?

jhass commented:
I'd also love to use this, how's the status?

kubecto commented:
What is going on now, can the arm64 image of k8s csi be used with a higher version？

Jean-Baptiste-Lasselle commented:
@motoki317 THANK YOU SOOOOO MUCH I JUST FINISHED TETING IT ALLLL AND IT WORKKSSSS on arm64 i run jupyterhub with that csi driver now with my own minio its just the heavennssss AWESOMEEE I really mean the THANK YOU

I had to trick around more things after that because i needed the official csi images like the registrar on ARM64 too, but i could easily find some from different companies on dockr hub with arm64 support, wow its really awesome ! without your work would have been muchhh harder!

motoki317 commented:
@Jean-Baptiste-Lasselle I’m glad you found it useful :)

Yes, you do need to swap some more CSI default images from what is provides in this repository, see the PR description and their latest versions to see how I did it.
> ref: [package](https://github.com/motoki317/k8s-csi-s3/pkgs/container/k8s-csi-s3), [manifest (values.yaml)](https://github.com/motoki317/manifest/blob/7a35d90de4229554ba01ad83b2d82e3024b633fe/csi-s3/values.yaml)

We should probably update those as well, maybe in another commit or PR.

schlichtanders commented:
Is this still being worked on?
I also would appreciate arm support

mertcangokgoz commented:
What about this PR, won't it support ARM?

denysvitali commented:
I've partially added support for this in #145

motoki317 commented:
> Sorry that you have to wait, I think your PR is fine, but I wasn't sure about the multi-architecture support in Yandex container registry. It seems it's not supported yet. So for now it won't work :) Container registry team says they are already working on a fix, so I'll merge it when it is ready.

@vitalif Hi, it's been a while, any update on this? It looks like a few other people also want this PR merged.

majkrzak commented:
Is there anything still blocking it? Are there any active forks implementing this maybe?
